worlds/pokemon_emerald/rules.py

```diff
@@ -0,0 +1,1366 @@
+            lambda state: state.has("HM02 Fly", player) and state.has("Feather Badge", player)
+        )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_LITTLEROOT_TOWN/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_LITTLEROOT_TOWN", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_OLDALE_TOWN/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_OLDALE_TOWN", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_PETALBURG_CITY/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_PETALBURG_CITY", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_RUSTBORO_CITY/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_RUSTBORO_CITY", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_DEWFORD_TOWN/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_DEWFORD_TOWN", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_SLATEPORT_CITY/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_SLATEPORT_CITY", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_MAUVILLE_CITY/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_MAUVILLE_CITY", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_VERDANTURF_TOWN/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_VERDANTURF_TOWN", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_FALLARBOR_TOWN/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_FALLARBOR_TOWN", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_LAVARIDGE_TOWN/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_LAVARIDGE_TOWN", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_FORTREE_CITY/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_FORTREE_CITY", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_LILYCOVE_CITY/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_LILYCOVE_CITY", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_MOSSDEEP_CITY/MAIN", player),
+        lambda state: state.has("EVENT_VISITED_MOSSDEEP_CITY", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_SOOTOPOLIS_CITY/EAST", player),
+        lambda state: state.has("EVENT_VISITED_SOOTOPOLIS_CITY", player)
+    )
+    set_rule(
+        multiworld.get_entrance("REGION_SKY -> REGION_EVER_GRANDE_CITY/SOUTH", player),
+        lambda state: state.has("EVENT_VISITED_EVER_GRANDE_CITY", player)
+    )
+
+    # Route 103
+    set_rule(
+        multiworld.get_entrance("REGION_ROUTE103/EAST -> REGION_ROUTE103/WATER", player),
+        can_surf
```

Comment:
your current _can_surf method can just be assigned here

Reply:
See above